### PR TITLE
feature: Novo Auth

### PR DESCRIPTION
**Pull request by icirellik**
This change rearranges the serverside middleware so that authentication
works correctly for the novo routes.

Followup:

- [x] Enabled `FlashMessage`
- [x] Swap Artsy font on Login Modal with SVG - https://github.com/artsy/force/pull/6716

### CODE REVIEW

File: src/desktop/lib/logoutHandler.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        window.analytics.reset()
+      }
+      if (redirectPath) {
+        location.assign(redirectPath)
+      } else {
+        location.reload()
+      }
+    })
+    .catch(errorMessage => {
+      // new FlashMessage({ message: errorMessage })
```
Comment (eessex) [inline src/desktop/lib/logoutHandler.ts:27]: We can use this error modal to display the message: https://github.com/artsy/force/blob/be156e9feda1505b3a231e6c3ea347d1a1a02cb9/src%2Fv2%2FComponents%2FModal%2FErrorModal.tsx#L18

File: src/lib/setup.ts [hunk 2]
```diff
@@ -156,9 +107,4 @@ export default function forceMiddleware(app) {
     // Mount desktop app
     app.use(require("../desktop"))
   }
-
-  // Ensure CurrentUser is set for Artsy Passport
-  // TODO: Investigate race condition b/t reaction's use of AP
-  // TODO: This is only run once, does it need to still happen
-  artsyPassport.options.CurrentUser = CurrentUser
```
Comment (eessex) [inline src/lib/setup.ts:163]: Does article authentication still work without this? That is the existing use case for reaction/passport.

File: package.json [hunk 3]
```diff
@@ -34,6 +34,7 @@
     "start:dev": "concurrently 'yarn start:dev:relay' 'yarn start:dev:server'",
     "start:dev:server": "NODE_ENV=development node --inspect --max_old_space_size=3072 -r dotenv/config ./src/index.dev.js",
     "start:dev:relay": "NODE_ENV=development yarn relay --watch",
+    "start:dev:debug": "node --inspect src/index.dev.js",
```
Comment (eessex) [inline package.json:37]: Intentional change for this pr? seems unrelated.

File: src/lib/setup.ts (hunk 2 above)
Comment (icirellik) [reply src/lib/setup.ts:163]: Yes

File: package.json (hunk 3 above)
Comment (icirellik) [reply package.json:37]: It was intentional, I will move it to another PR because I already have 😄

File: src/novo/src/server.ts [hunk 4]
```diff
@@ -113,14 +117,18 @@ function initializeNovo() {
...
+              <div id='react-root'>${bodyHTML}</div>
 
               ${scripts}
+              <style type="text/css">
+              .grecaptcha-badge { visibility: hidden; }
+              </style>
               <script src="${novoManifest.lookup(
                 "/assets-novo/novo-artsy-novo.js"
               )}"></script>
+              <!-- TODO: add eigen exclude -->
```
Comment (eessex) [inline src/novo/src/server.ts:130]: Still need to exclude this script from loading in eigen, can use var `sd.EIGEN`

Comment (icirellik) [reply src/novo/src/server.ts:130]: I would prefer to wait until the current HTML Webpack plugin task is in place. The template will require redoing this work from scratch.

Comment (icirellik) [reply src/novo/src/server.ts:130]: https://github.com/artsy/force/pull/6741/files#diff-8e91e6a62be6cbd001214e39acfed4a04d076583afcc37ef3152e5f5cfca1c65R25
